server/server.go

```diff
@@ -1993,10 +1993,16 @@ func (s *Server) checkPathScope(ctx context.Context, txn storage.Transaction, pa
 	}
 
 	spath := strings.Trim(path.String(), "/")
+	spathBase := strings.Split(spath, "/")[0]
+
+	if spath == "" && len(bundleRoots) > 0 {
+		return types.BadRequestErr("can't write to document root with bundle roots configured")
+	}
 
 	for name, roots := range bundleRoots {
 		for _, root := range roots {
-			if strings.HasPrefix(spath, root) || strings.HasPrefix(root, spath) {
+			rootBase := strings.Split(root, "/")[0]
+			if spathBase == rootBase {
```

Comment:
I might just be confused (and that's rather likely), but would we not end up being too strict here?

```golang
	spathBase := strings.Split("foo/bar/baz", "/")[0]
	rootBase := strings.Split("foo/bar/quz", "/")[0]
	fmt.Println(spathBase, rootBase, spathBase == rootBase)
```
This prints `[foo foo true]`, but the paths don't actually overlap.

Reply:
Yep, should have marked as WIP - working on taking each path component into account now.